evm/test/forks/HubMainnetFork.t.sol

```diff
@@ -0,0 +1,373 @@
+// SPDX-License-Identifier: Apache 2
+pragma solidity ^0.8.23;
+
+import {Test, console} from "forge-std/Test.sol";
+import {Vm} from "forge-std/Vm.sol";
+
+// Hub Contracts
+import {TimelockController} from "@openzeppelin/contracts/governance/TimelockController.sol";
+import {HubGovernor} from "src/HubGovernor.sol";
+import {HubProposalExtender} from "src/HubProposalExtender.sol";
+import {HubVotePool} from "src/HubVotePool.sol";
+import {HubProposalMetadata} from "src/HubProposalMetadata.sol";
+import {HubMessageDispatcher} from "src/HubMessageDispatcher.sol";
+import {HubEvmSpokeAggregateProposer} from "src/HubEvmSpokeAggregateProposer.sol";
+import {HubSolanaMessageDispatcher} from "src/HubSolanaMessageDispatcher.sol";
+import {HubSolanaSpokeVoteDecoder} from "src/HubSolanaSpokeVoteDecoder.sol";
+import {ERC20Votes} from "@openzeppelin/contracts/token/ERC20/extensions/ERC20Votes.sol";
+import {Governor} from "@openzeppelin/contracts/governance/Governor.sol";
+import {IGovernor} from "@openzeppelin/contracts/governance/IGovernor.sol";
+
+contract HubMainnetForkTest is Test {
+  string ETHEREUM_RPC_URL = vm.envString("ETHEREUM_RPC_URL");
+  uint256 ethereumForkId;
+
+  // Deployed Contract Addresses (from mainnet-test-deploy-contracts.md)
+  address constant TIMELOCK_ADDR = 0x0fAA8fc7A60809B3557d5Dbe463B64F94de5ac06;
+  address constant EXTENDER_ADDR = 0xB85D4a7D0661Afa0EFEFF9E3B6Fc82bf427e6C69;
+  address constant HUB_VOTE_POOL_ADDR = 0x6D87469dC04aec896dB03Df9B1b9Ba29535CC206;
+  address constant GOV_ADDR = 0x50b97697DbDa7a38f249966E02CCE6064657c54B;
+  address constant HUB_EVM_VOTE_DECODER_ADDR = 0xa891e332E50E2d3105a5F9b85cFBbFc1D8E05541;
+  address constant HUB_SOLANA_VOTE_DECODER_ADDR = 0xc70cc0f137fA23b5A42EE48a9A48E52c345E8dA4;
+  address constant HUB_METADATA_ADDR = 0xe1485b53e6E94aD4B82b19E48DA1911d2E19bFaE;
+  address constant HUB_MSG_DISPATCHER_ADDR = 0xb2F162945eF0631F62FE4421dc6Ec5eCDf92EF59;
+  address constant HUB_SOLANA_DISPATCHER_ADDR = 0xadB8de6dfB41a1Fce6635460E77bEaDc73148BE4;
+  address constant HUB_EVM_AGG_PROPOSER_ADDR = 0xb2490491FBb846B314D3ce65D77f9f27Ef964b4F;
+  address constant TEST_WTOKEN_ADDR = 0x691d45404441c4a297ecCc8dE29C033afCeaac3e;
+  // TODO: Replace with actual WToken address for production verification (delete the above)
+  address constant W_TOKEN_ADDR = 0xB0fFa8000886e57F86dd5264b9582b2Ad87b2b91;
+
+  // Testnet Spoke Addresses & Chain IDs (from mainnet-test-deploy-contracts.md & scripts)
+  address constant ARBITRUM_SPOKE_AGG_ADDR = 0x6dEfA659A9726925307a45B30Ffe2Da45ED90811;
+  uint16 constant ARBITRUM_CHAIN_ID = 23;
+  address constant BASE_SPOKE_AGG_ADDR = 0x31eD7EAa0CCA7e95a93339843a1C257b87e31E3d;
+  uint16 constant BASE_CHAIN_ID = 30;
+  address constant OPTIMISM_SPOKE_AGG_ADDR = 0x75F755950D59d2007A0C90457fDc190732567cC5;
+  uint16 constant OPTIMISM_CHAIN_ID = 24;
```

Comment:
Might be cleaner to move these to a constants file

Reply:
https://github.com/wormhole-foundation/multigov/pull/268/commits/0d683469ea970fc902f17166937249e33c5eece4